loki/transform/transform_hoist_variables.py

```diff
@@ -150,6 +151,8 @@ def transform_subroutine(self, routine, **kwargs):
         call_map = CaseInsensitiveDict((str(call.name), call) for call in calls)
 
         for child in successors:
+            if not isinstance(child, SubroutineItem):
+                continue
             arg_map = dict(call_map[child.local_name].arg_iter())
```

Comment:
If SCC HOIST is used together with `GlobalVarOffloadTransformation`, some of `successors` may be `Item`s (corresponding to the imported global variables). Therefore, adding this if statement to skip processing for `Item`s other than `SubroutineItem`s. To me, everything here seems to indicate this loop only processes subroutines.